website/docs/feature-flag-tutorials/use-cases/trunk-based-development.md

```diff
@@ -0,0 +1,222 @@
+
+### Microservices vs. Monoliths
+
+The underlying application architecture can also influence the optimal trunk-based development strategy. Teams working on monolithic applications may find more success with less frequent merges, while those working on microservices may benefit from a more continuous integration approach.
+
+**Monolithic Applications**:
+
+-   Larger, more tightly coupled codebase
+-   Merging at set intervals (e.g., daily) may be more suitable to manage integration complexity
+-   Increased emphasis on rigorous testing and staged deployments
+
+**Microservices**:
+
+-   Smaller, more loosely coupled services
+-   Merging every commit is often more feasible due to reduced integration challenges
+-   Enables more independent development and deployment of individual services
+
+## Create Feature Flags for Incomplete Features
+
+Some of the most appropriate feature candidates for trunk-based development are features with significant potential impact on users that may require thorough testing.
+
+In TBD, developers merge incomplete code into the main branch. Unleash feature flags allow this to happen safely by keeping the feature hidden or disabled, ensuring the trunk remains in a releasable state. The flag starts as “off” in production while the feature is incomplete. The flag can be on in development and/or a testing environment until the feature is complete and ready to go live in production for all users.
+
+We’ll create a flag that will wrap around an incomplete feature in your application code. After that, we’ll explore rollout strategies and how they are configured in Unleash to enable trunk-based development.
+
+In the Unleash Admin UI, open a project and click **New feature flag**.
+
+![Create a new feature flag in the Unleash Admin UI.](/img/use-case-new-flag.png)
+
+Next, name the new feature flag. The purpose of this flag is to manage the deployment of a new or incomplete feature. Therefore, the flag we are creating is considered a Release flag type. While release flags are invaluable, they should be short-lived to prevent codebase complexity and technical debt accumulation.
+
+![Create a feature flag by filling out the form fields.](/img/use-case-create-tbd-flag.png)
+
+Once you have completed the form, click **Create feature flag**.
+
+Once your flag is created, you can build a feature in your application code while keeping it hidden in production.
+
+### Effectively Manage your Feature Flag and Code
+
+When implementing trunk-based development with Unleash, it's crucial to have a well-structured process for managing your feature flags and the associated code. This ensures the trunk remains in a deployable state while allowing teams to work independently on different features.
+
+#### Establish Consistent Naming Conventions
```

Comment:
Worth linking to our docs on flag [naming patterns](https://docs.getunleash.io/reference/feature-toggles#set-a-naming-pattern)?